content/francineblanc/how-to-use-git-for-beginners.md

```diff
@@ -0,0 +1,219 @@
+
+If the rewind needs to go beyond the most recent commit, the command ```git reset SHA```, where ```SHA``` is the first 7 charcters of the SHA of the commit, can be used. Note that the SHA of all previous commits can be viewed using ```git log```.
+
+More about the ```reset``` command can be viewed in the [Reset] section of the Codecademy docs for Git.
+
+## Git Branching
+In the Initialising Git section of this article, the output from running ```git status``` referred to being on ```branch main```. In Git, branches are a core feature. They can be created, deleted, compared, merged and tracked. 
+
+The first branch to be aware of is the ```HEAD``` branch, which is the currently active branch. So, the ```main``` branch referenced above was the currently active branch and in GitHub is the name of the default branch. In Git, the default branch name is ```master```, which was also originally the case in GitHub however this was renamed for the reasons outlined [here]. The main take away though is that both ```master``` and ```main``` are used as default branch names.
+
+The idea of Git branches is that different branches can be created and worked on, and changes can be merged into the ```main``` branch. Branches are in effect an independent line for the adding/staging/committing process, forked from the ```main``` branch. Once on a branch, commits are recorded in that branch's history and when ready a branch (including the changes made on that branch) can be merged i.e. combined into ```main```.
+
+A new branch can be made with the following command:
+```
+git branch <name-of-new-branch>
+```
+To switch over to the new branch in a project the command:
+```
+git checkout <name-of-the-branch>
+```
+is used. These two steps can however be neatly combined into one using:
+```
+git checkout -b <name-of-new-branch>
+```
+Once on a new branch, files in the Working Directory can be worked on as usual, and changes can be added and committed. Those changes will however only be committed to the current branch and will not affect anything on ```main```.
+
+### Merging
+Merging combines multiple commits into one history, and is generally used to join or combine two branches. So, when ready for work on a branch to be merged into ```main```, the command ```git merge branch-name``` is used, where ```branch-name``` is the name of the branch to be merged into the ```main``` branch. This command is most often used when working on a project with others (more on this below).
+
+Once a branch has been merged, it is no longer needed so can be deleted with:
+
+```
+git branch -d <branch-name>
+``` 
+This will though only delete the branch once it has been merged into ```main```. Replacing the ```-d``` flag with ```-D``` will delete the branch even if it hasn't been merged with ```main```.
```

Comment:
```suggestion

This will only delete the branch once it has been merged into the `main` branch. Replacing the `-d` flag with `-D` will force-delete the branch.
```